aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/MutationProcessor.java

```diff
@@ -179,24 +187,25 @@ void stopDrainingMutationOutbox() {
      */
     private <T extends Model> Single<ModelWithMetadata<T>> publishWithStrategy(
             PendingMutation<T> mutation, PublicationStrategy<T> publicationStrategy) {
-        return Single.defer(() -> Single.create(subscriber -> {
+        T mutatedItem = mutation.getMutatedItem();
+        String modelClassName = mutation.getClassOfMutatedItem().getSimpleName();
+        return Single.defer(() -> Single.create(subscriber ->
             publicationStrategy.publish(
-                mutation.getMutatedItem(),
+                mutatedItem,
                 result -> {
                     if (!result.hasErrors() && result.hasData()) {
                         subscriber.onSuccess(result.getData());
                         return;
                     }
-                    String modelName = mutation.getClassOfMutatedItem().getSimpleName();
                     subscriber.onError(new DataStoreException(
                         "Mutation failed. Failed mutation = " + mutation + ". " +
                             "AppSync response contained errors = " + result.getErrors(),
-                        "Verify that your AppSync endpoint is able to store " + modelName + " models."
+                        "Verify that your AppSync endpoint is able to store " + modelClassName + " models."
```

Comment:
I've found this recovery suggestion dubious thus far (thought I don't have a suggestion here).

Reply:
I'm not sold on the recovery suggestion thing, overall, honestly. We're either not trying hard enough, or, it was a well-intentioned idea that doesn't make sense in practice.

Reply:
I’ve always liked the idea of exceptions that lead to wiki pages that explained what happened with links to code, and where people can document issues, workarounds, gotchas. Maybe one day. :-)